server.php

```diff
@@ -17,6 +17,38 @@ function show_valet_404()
 }
 
 /**
+ * http://stackoverflow.com/a/10473026/2098861
+ *
+ * @param $haystack
+ * @param $needle
+ *
+ * @return bool
+ */
+function ends_with($haystack, $needle)
+{
+    // search forward starting from end minus needle length characters
+    return $needle === '' || (($temp = strlen($haystack) - strlen($needle)) >= 0 && strpos($haystack, $needle, $temp) !== false);
+}
+
+/**
+ * @param $domain string Domain to filter
+ *
+ * @return string Filter domain (without xip.io feature)
+ */
+function support_xip_io($domain)
+{
+    if (ends_with($domain, ".xip.io")) {
```

Comment:
One other thing I just realized, we try really hard not to define functions that could cause collisions in `server.php` because this file is included on every single request, and if the site being served tries to define a function we've already defined, the user will get an error.

Could we just inline a hardcoded implementation here instead since it's only used once?

`if (substr($domain, -7) === '.xip.io') {`

Thanks!